manuals/evtinfo.1
.Dd June 25, 2026
.Dt EVTINFO 1
.Os
.Sh NAME
.Nm evtinfo
.Nd determines information about a Windows Event Log (EVT) file
.Sh SYNOPSIS
.Nm evtinfo
.Op Fl c Ar codepage
.Op Fl hvV
.Ar source
.Sh DESCRIPTION
.Nm evtinfo
is a utility to determine information about a Windows Event Log (EVT) file
.Pp
.Nm evtinfo
is part of the
.Nm libevt
package.
.Nm libevt
is a library to access the Windows Event Log (EVT) format
.Pp
.Ar source
is the source file.
.Pp
The options are as follows:
.Bl -tag -width Ds
.It Fl c Ar codepage
codepage of ASCII strings, options: ascii, windows-874, windows-932, \
windows-936, windows-949, windows-950, windows-1250, windows-1251, windows-1252 \
(default), windows-1253, windows-1254, windows-1255, windows-1256, windows-1257 \
or windows-1258
.It Fl h
shows this help
.It Fl v
verbose output to stderr
.It Fl V
print version
.El
.Sh ENVIRONMENT
None
.Sh FILES
None
.Sh EXAMPLES
.Bd -literal
# evtinfo AppEvent.Evt
evtinfo 20260625
.sp
Windows Event Log (EVT) information:
	Version				: 1.1
	Number of records		: 19
	Number of recovered records	: 0
	Log type			: Application
	Flags:
		Should be archived
.sp
.Ed
.Sh DIAGNOSTICS
Errors, verbose and debug output are printed to stderr when verbose output \
\-v is enabled.
Verbose and debug output are only printed when enabled at compilation.
.Sh SEE ALSO
.Xr evtexport 1
.Sh AUTHORS
.An Joachim Metz <joachim.metz@gmail.com>
.Sh BUGS
Please report bugs of any kind on the project issue tracker: \
https://github.com/libyal/libevt/issues
.Sh COPYRIGHT
Copyright (C) 2011-2026, Joachim Metz <joachim.metz@gmail.com>.
.sp
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A \
PARTICULAR PURPOSE.


ChangeLog
TODO:
* move direct access to event record struct members in record values to functions
  - replace calls to libfvalue by libuna and libfwnt
  - handle strings, use split string?
* API have source and computer name functions return 0

* libevt
  - remove file_get_version
* evtexport, add date time format selection
* evtexport, add flags to strip strings and messages string of CR-LFs
* evtexport, add support for CVS and/or TSV format with surrounding quotes

* add parameter expansion support
* recovery: what about record that is recoverable but has a size mismatch
* msvscpp:
  - check and fix debug output generation
  - check 64-bit compilation output (warnings)
* check signal abort
* test find paths on windows
* get %WinDir%
  SYSTEM\CurrentControlSet\Control\Session Manager\Environment\windir
  - add callback for dealing with path expansion ?
* registry file:
  - detect the file type + version based on the base sub keys?
  - add file type specific support e.g. CurrentControlSet in key names ?
    or be more generic and allow HKEY_ ... type of keys and detect
    registry file type

* libcdirectory:
  - move WINAPI * path expansion from export handle to libcdirectory
* update manuals
  - evtexport add example output
* add error tollerability (+/-)
  - sizes and copy

Tools:
* API
  - functionality to parse event identifier ?
* registy file
  - what about NT4 root/base key support ?
* evtexport/export handle
  - validate eventlog/computername in system registry ?
  - use current control set nr ?
  - print if message file key not found
  - print if message file not found
  - print if message file has no resources
  - user sid - find corresponding user name ?
  - output data as CSV/TSV, DFXML ? (different output formatting modules)
  - print data
  - move output to separate code, e.g. libpff
* evtinfo
* pyevt
  - functionality to parse event identifier ?
  - string representations of event type ?
  - access to data
* tests
  - file wrap

Format:
* event flags ? (debug function)
* closing record number ?

Notes
* %SystemRoot% expansion support for WINNT and WINNT35
  check if this is set in SOFTWARE\Microsoft\Windows NT\CurrentVersion\ ?

20111003
* see `git log' for more recent change log
* Created initial version based on libnk2



evttools/evtinfo.c
/*
 * Shows information obtained from a Windows Event Log (EVT) file.
 *
 * Copyright (C) 2011-2026, Joachim Metz <joachim.metz@gmail.com>
 *
 * Refer to AUTHORS for acknowledgements.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

#include <common.h>
#include <file_stream.h>
#include <system_string.h>
#include <types.h>

#if defined( HAVE_FCNTL_H ) || defined( WINAPI )
#include <fcntl.h>
#endif

#if defined( HAVE_IO_H ) || defined( WINAPI )
#include <io.h>
#endif

#if defined( HAVE_STDLIB_H ) || defined( WINAPI )
#include <stdlib.h>
#endif

#if defined( HAVE_UNISTD_H )
#include <unistd.h>
#endif

#include "evttools_getopt.h"
#include "evttools_libcerror.h"
#include "evttools_libclocale.h"
#include "evttools_libcnotify.h"
#include "evttools_libevt.h"
#include "evttools_output.h"
#include "evttools_signal.h"
#include "evttools_unused.h"
#include "info_handle.h"

info_handle_t *evtinfo_info_handle = NULL;
int evtinfo_abort                  = 0;

/* Signal handler for evtinfo
 */
void evtinfo_signal_handler(
      evttools_signal_t signal EVTTOOLS_ATTRIBUTE_UNUSED )
{
	libcerror_error_t *error = NULL;
	static char *function    = "evtinfo_signal_handler";

	EVTTOOLS_UNREFERENCED_PARAMETER( signal )

	evtinfo_abort = 1;

	if( evtinfo_info_handle != NULL )
	{
		if( info_handle_signal_abort(
		     evtinfo_info_handle,
		     &error ) != 1 )
		{
			libcnotify_printf(
			 "%s: unable to signal info handle to abort.\n",
			 function );

			libcnotify_print_error_backtrace(
			 error );
			libcerror_error_free(
			 &error );
		}
	}
	/* Force stdin to close otherwise any function reading it will remain blocked
	 */
#if defined( WINAPI ) && !defined( __CYGWIN__ )
	if( _close(
	     0 ) != 0 )
#else
	if( close(
	     0 ) != 0 )
#endif
	{
		libcnotify_printf(
		 "%s: unable to close stdin.\n",
		 function );
	}
}

/* The main program
 */
#if defined( HAVE_WIDE_SYSTEM_CHARACTER )
int wmain( int argc, wchar_t * const argv[] )
#else
int main( int argc, char * const argv[] )
#endif
{
	const char *description = \
		"Use evtinfo to determine information about a Windows Event Log (EVT) file.";

	evttools_option_t options[ ] = {
		{ 'c', "codepage", "codepage of ASCII strings, options: ascii, windows-874, windows-932, windows-936, windows-949, windows-950, windows-1250, windows-1251, windows-1252 (default), windows-1253, windows-1254, windows-1255, windows-1256, windows-1257 or windows-1258" },
		{ 'h', NULL, "shows this help" },
		{ 'v', NULL, "verbose output to stderr" },
		{ 'V', NULL, "print version" },
		{ 0, "source", "the source file" },
	};
	system_character_t options_string[ 32 ];

	int result                          = 0;
	libevt_error_t *error               = NULL;
	system_character_t *option_codepage = NULL;
	system_character_t *source          = NULL;
	char *program                       = "evtinfo";
	system_integer_t option             = 0;
	int number_of_options               = (int) ( sizeof( options ) / sizeof( evttools_option_t ) );
	int verbose                         = 0;

#if defined( __MINGW32__ ) && defined( HAVE_MINGW_BINMODE )
	_setmode( _fileno( stdout ), _O_BINARY );
	_setmode( _fileno( stderr ), _O_BINARY );
#endif

	libcnotify_stream_set(
	 stderr,
	 NULL );
	libcnotify_verbose_set(
	 1 );

	if( libclocale_initialize(
	     "evttools",
	     &error ) != 1 )
	{
		fprintf(
		 stderr,
		 "Unable to initialize locale values.\n" );

		goto on_error;
	}
	if( evttools_output_initialize(
	     _IONBF,
	     &error ) != 1 )
	{
		fprintf(
		 stderr,
		 "Unable to initialize output settings.\n" );

		goto on_error;
	}
	evttools_output_version_fprint(
	 stdout,
	 program );

	if( evttools_getopt_get_options_string(
	     options,
	     number_of_options,
	     options_string,
	     32 ) != 1 )
	{
		fprintf(
		 stderr,
		 "Unable to determine options string.\n" );

		goto on_error;
	}
	while( ( option = evttools_getopt(
	                   argc,
	                   argv,
	                   options_string ) ) != (system_integer_t) -1 )
	{
		switch( option )
		{
			case (system_integer_t) '?':
			default:
				fprintf(
				 stderr,
				 "Invalid argument: %" PRIs_SYSTEM "\n",
				 argv[ optind - 1 ] );

				evttools_getopt_usage_fprint(
				 stdout,
				 program,
				 description,
				 options,
				 number_of_options );

				return( EXIT_FAILURE );

			case (system_integer_t) 'c':
				option_codepage = optarg;

				break;

			case (system_integer_t) 'h':
				evttools_getopt_usage_fprint(
				 stdout,
				 program,
				 description,
				 options,
				 number_of_options );

				return( EXIT_SUCCESS );

			case (system_integer_t) 'v':
				verbose = 1;

				break;

			case (system_integer_t) 'V':
				evttools_output_copyright_fprint(
				 stdout );

				return( EXIT_SUCCESS );
		}
	}
	if( optind == argc )
	{
		fprintf(
		 stderr,
		 "Missing source file.\n" );

		evttools_getopt_usage_fprint(
		 stdout,
		 program,
		 description,
		 options,
		 number_of_options );

		return( EXIT_FAILURE );
	}
	source = argv[ optind ];

	libcnotify_verbose_set(
	 verbose );
	libevt_notify_set_stream(
	 stderr,
	 NULL );
	libevt_notify_set_verbose(
	 verbose );

	if( info_handle_initialize(
	     &evtinfo_info_handle,
	     &error ) != 1 )
	{
		fprintf(
		 stderr,
		 "Unable to initialize info handle.\n" );

		goto on_error;
	}
	if( option_codepage != NULL )
	{
		result = info_handle_set_ascii_codepage(
		          evtinfo_info_handle,
		          option_codepage,
		          &error );

		if( result == -1 )
		{
			fprintf(
			 stderr,
			 "Unable to set ASCII codepage in info handle.\n" );

			goto on_error;
		}
		else if( result == 0 )
		{
			fprintf(
			 stderr,
			 "Unsupported ASCII codepage defaulting to: windows-1252.\n" );
		}
	}
	result = info_handle_set_event_log_type_from_filename(
	          evtinfo_info_handle,
	          source,
	          &error );

	if( result == -1 )
	{
		fprintf(
		 stderr,
		 "Unable to set event log type from filename in info handle.\n" );

		goto on_error;
	}
	if( info_handle_open_input(
	     evtinfo_info_handle,
	     source,
	     &error ) != 1 )
	{
		fprintf(
		 stderr,
		 "Unable to open source file.\n" );

		goto on_error;
	}
	if( info_handle_file_fprint(
	     evtinfo_info_handle,
	     &error ) != 1 )
	{
		fprintf(
		 stderr,
		 "Unable to print file information.\n" );

		goto on_error;
	}
	if( info_handle_close_input(
	     evtinfo_info_handle,
	     &error ) != 0 )
	{
		fprintf(
		 stderr,
		 "Unable to close info handle.\n" );

		goto on_error;
	}
	if( info_handle_free(
	     &evtinfo_info_handle,
	     &error ) != 1 )
	{
		fprintf(
		 stderr,
		 "Unable to free info handle.\n" );

		goto on_error;
	}
	return( EXIT_SUCCESS );

on_error:
	if( error != NULL )
	{
		libcnotify_print_error_backtrace(
		 error );
		libcerror_error_free(
		 &error );
	}
	if( evtinfo_info_handle != NULL )
	{
		info_handle_free(
		 &evtinfo_info_handle,
		 NULL );
	}
	return( EXIT_FAILURE );
}

